### PR TITLE
Filter documents using Go code

### PR DESCRIPTION
**Pull request by AlekSi**

### CODE REVIEW

File: internal/handlers/common/projection.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import "github.com/FerretDB/FerretDB/internal/types"
+
+// ProjectDocuments modifies given documents in places according to the given projection.
+func ProjectDocuments(docs []*types.Document, projection *types.Document) error {
+	// TODO
```
Comment (AlekSi) [inline internal/handlers/common/projection.go:21]: Missing

File: internal/handlers/common/sort.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import "github.com/FerretDB/FerretDB/internal/types"
+
+// SortDocuments sorts given documents in place according to the given sorting conditions.
+func SortDocuments(docs []*types.Document, sort *types.Document) error {
+	// TODO
```
Comment (AlekSi) [inline internal/handlers/common/sort.go:21]: Missing

File: internal/handlers/jsonb1/msg_update.go [hunk 3]
```diff
@@ -18,159 +18,154 @@ import (
...
-			"nModified", updated,
-			"ok", float64(1),
-		)},
-	})
-	if err != nil {
-		return nil, lazyerrors.Error(err)
-	}
-
-	return &reply, nil
+	return nil, fmt.Errorf("TODO")
```
Comment (AlekSi) [inline internal/handlers/jsonb1/msg_update.go:26]: Missing

File: internal/handlers/common/projection.go (hunk 1 above)
Comment (AlekSi) [reply internal/handlers/common/projection.go:21]: @seeforschauer

File: internal/handlers/common/sort.go (hunk 2 above)
Comment (AlekSi) [reply internal/handlers/common/sort.go:21]: @w84thesun

File: internal/handlers/common/compare.go [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+		panic("a is nil")
+	}
+	if b == nil {
+		panic("b is nil")
+	}
+
+	switch a := a.(type) {
+	case float64:
+		switch b := b.(type) {
+		case float64:
```
Comment (seeforschauer) [inline internal/handlers/common/compare.go:49]: what about float32?

File: internal/handlers/common/projection.go (hunk 1 above)
Comment (seeforschauer) [reply internal/handlers/common/projection.go:21]: :ok_hand:

File: internal/handlers/common/storage.go [hunk 5]
```diff
@@ -21,9 +21,10 @@ import (
 )
 
 type Storage interface {
+	MsgCount(context.Context, *wire.OpMsg) (*wire.OpMsg, error)
 	MsgCreateIndexes(context.Context, *wire.OpMsg) (*wire.OpMsg, error)
 	MsgDelete(context.Context, *wire.OpMsg) (*wire.OpMsg, error)
-	MsgFindOrCount(context.Context, *wire.OpMsg) (*wire.OpMsg, error)
```
Comment (seeforschauer) [inline internal/handlers/common/storage.go:26]: :+1:

File: internal/handlers/jsonb1/msg_find.go [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+import (
+	"context"
+
+	"github.com/FerretDB/FerretDB/internal/handlers/common"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+	"github.com/FerretDB/FerretDB/internal/wire"
+)
+
+// MsgFindOrCount finds documents in a collection or view and returns a cursor to the selected documents
```
Comment (seeforschauer) [inline internal/handlers/jsonb1/msg_find.go:26]:
```suggestion
// MsgFind finds documents in a collection or view and returns a cursor to the selected documents
```

File: internal/handlers/jsonb1/msg_count.go [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+import (
+	"context"
+
+	"github.com/FerretDB/FerretDB/internal/handlers/common"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+	"github.com/FerretDB/FerretDB/internal/wire"
+)
+
+// MsgFindOrCount finds documents in a collection or view and returns a cursor to the selected documents
```
Comment (seeforschauer) [inline internal/handlers/jsonb1/msg_count.go:26]:
```suggestion
// MsgCount returns the count of documents that's matched by the query
```

File: internal/handlers/jsonb1/msg_count.go [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+		}
+
+		if !matches {
+			continue
+		}
+
+		resDocs = append(resDocs, doc)
+	}
+
+	if err = common.SortDocuments(resDocs, sort); err != nil {
```
Comment (seeforschauer) [inline internal/handlers/jsonb1/msg_count.go:103]: Is it necessary to sort documents on count query?

File: internal/handlers/jsonb1/msg_find.go [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+			return nil, err
+		}
+	}
+
+	var reply wire.OpMsg
+	err = reply.SetSections(wire.OpMsgSection{
+		Documents: []*types.Document{types.MustNewDocument(
+			"cursor", types.MustNewDocument(
+				"firstBatch", firstBatch,
+				"id", int64(0), // TODO
```
Comment (seeforschauer) [inline internal/handlers/jsonb1/msg_find.go:128]: what TODO is meant?

File: internal/handlers/common/compare.go (hunk 4 above)
Comment (AlekSi) [reply internal/handlers/common/compare.go:49]: float32 is not a BSON type

Comment (AlekSi) [reply internal/handlers/common/compare.go:49]: See types package and https://bsonspec.org/spec.html

File: internal/handlers/jsonb1/msg_count.go (hunk 8 above)
Comment (AlekSi) [reply internal/handlers/jsonb1/msg_count.go:103]: No. Nice catch!

File: internal/handlers/jsonb1/msg_find.go (hunk 9 above)
Comment (AlekSi) [reply internal/handlers/jsonb1/msg_find.go:128]: Cursor support. Not for this PR

**Pull request merged**